src/bic/view/form/controls.js

```diff
@@ -200,11 +196,10 @@ define(function (require) {
         formErrors = currentForm.getInvalidElements();
         this.addToQueue(formErrors && formErrors.length ? MODEL_STATUS.DRAFT : MODEL_STATUS.PENDING)
             .then(leaveViewBy(this, USER_ACTIONS.SUBMIT))
-            // CATCH - enable submit if there has been an error
-            .then(undefined, function (invalidElements) {
+            .catch(function (invalidElements) {
               me.model.trigger('showErrors');
               enableSubmit();
-              scrollToFirstError(invalidElements);
+              currentForm.get('_view').goToElement(invalidElements.errors[0]);
```

Comment:
yes, the promise is only rejected on line 333 in the same file. The only time it will fail is if we change the output of `currentForm#getInvalidElements`, but if we do that quite a lot of things will be broken